app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityUnregisterFromActivityTest.java

```diff
@@ -0,0 +1,186 @@
+package com.sdp.movemeet.view.activity;
+
+import android.content.Intent;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.storage.StorageReference;
+import com.sdp.movemeet.backend.BackendManager;
+import com.sdp.movemeet.backend.firebase.firestore.FirestoreActivityManager;
+import com.sdp.movemeet.backend.providers.AuthenticationInstanceProvider;
+import com.sdp.movemeet.backend.serialization.ActivitySerializer;
+import com.sdp.movemeet.models.Activity;
+import com.sdp.movemeet.models.Sport;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.concurrent.CountDownLatch;
+
+import static androidx.test.core.app.ApplicationProvider.getApplicationContext;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+@RunWith(AndroidJUnit4.class)
+public class ActivityDescriptionActivityUnregisterFromActivityTest {
```

Comment:
Same question as before

Reply:
Yes this is basically the same principle as the previous test. Here I test that the user is properly unregistered from the activity to which he registered. More precisely, the first part of the test consists of first registering the user to this activity and the second part to unregister him (i.e. remove his `userId` String from the `participantId` array field of the activity document in Firebase Firestore).